File: 713_Subarray_Product_Less_Than_K/sliding_window.c
int numSubarrayProductLessThanK(int* nums, int numsSize, int k){
    if (k == 0 || numsSize <= 0){
        return 0;
    }
    int l = 0;
    int r = 0;
    int prod = 1;
    int count = 0;
    for (r = 0; r < numsSize; r++){
        prod *= nums[r];
        while (prod >= k && l <= r){
            prod /= nums[l];
            l += 1;
        }
        printf("r = %d, l = %d\n", r,l);
        count += r-l+1;
    }
    return count;
}

File: Leetcode_problems/713_Subarray_Product_Less_Than_K/sliding_window.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k == 0 or len(nums)== 0:
            return 0

        l = 0
        r = 0
        prod = 1
        count = 0
        for r in range(len(nums)):
            prod *= nums[r]
            while prod >= k and l<=r:
                prod /= nums[l]
                l += 1
            count += r-l+1
        return count
